src/mpp/Dec.hpp

```diff
@@ -1172,6 +1172,27 @@ bool jump_read_key(BUF& buf, T... t)
 	return jump_find_key<PAIRS, FAMILY, PATH>(val, IS, buf, t...);
 }
 
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF, class = void>
+struct is_optional_or_variant_readable_in_place : std::false_type {};
+
+template<compact::Family FAMILY, size_t SUBRULE,
+	 class DST, class BUF>
+struct is_optional_or_variant_readable_in_place
+	<FAMILY, SUBRULE, DST, BUF,
+	 typename std::enable_if_t<(FAMILY == compact::MP_BOOL ||
+				    FAMILY == compact::MP_INT ||
+				    FAMILY == compact::MP_FLT) &&
+				   std::is_assignable_v<DST,
+				   decltype(read_value<FAMILY,
+					    SUBRULE>(std::declval<BUF>()))>,
+					    bool>::type> : std::true_type {};
```

Comment:
It was difficult to understand this snippet, probably, it's better to change alignment.

Reply:
Fixed.